internal/webhooks/cluster.go

```diff
@@ -259,41 +240,32 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 		)
 	}
 
-	// clusterClass must exist.
-	clusterClass := &clusterv1.ClusterClass{}
-	// Check to see if the ClusterClass referenced in the Cluster currently exists.
-	if err := webhook.Client.Get(ctx, client.ObjectKey{Namespace: newCluster.Namespace, Name: newCluster.Spec.Topology.Class}, clusterClass); err != nil {
+	// Get the ClusterClass referenced in the Cluster.
+	clusterClass, clusterClassGetErr := webhook.pollClusterClassForCluster(ctx, newCluster)
+	if clusterClassGetErr == nil {
+		// If there's no error validate the Cluster based on the ClusterClass.
+		allErrs = append(allErrs, ValidateClusterForClusterClass(newCluster, clusterClass, fldPath)...)
+	} else if !apierrors.IsNotFound(clusterClassGetErr) {
+		// If the error is anything other than "Not Found" return all errors at this point.
 		allErrs = append(
-			allErrs, field.Invalid(
+			allErrs, field.InternalError(
 				fldPath.Child("class"),
-				newCluster.Name,
-				fmt.Sprintf("ClusterClass with name %q could not be found", newCluster.Spec.Topology.Class)))
+				clusterClassGetErr))
 		return allErrs
 	}
 
-	allErrs = append(allErrs, check.MachineDeploymentTopologiesAreValidAndDefinedInClusterClass(newCluster, clusterClass)...)
-
-	// Check if the variables defined in the ClusterClass are valid.
-	allErrs = append(allErrs, variables.ValidateClusterVariables(newCluster.Spec.Topology.Variables, clusterClass.Spec.Variables,
-		fldPath.Child("variables"))...)
-
-	// validate the MachineHealthChecks defined in the cluster topology
-	allErrs = append(allErrs, validateMachineHealthChecks(newCluster, clusterClass)...)
-
-	if newCluster.Spec.Topology.Workers != nil {
-		for i, md := range newCluster.Spec.Topology.Workers.MachineDeployments {
-			// Continue if there are no variable overrides.
-			if md.Variables == nil || len(md.Variables.Overrides) == 0 {
-				continue
-			}
-
-			allErrs = append(allErrs, variables.ValidateMachineDeploymentVariables(md.Variables.Overrides, clusterClass.Spec.Variables,
-				fldPath.Child("workers", "machineDeployments").Index(i).Child("variables", "overrides"))...)
+	if oldCluster != nil { // On update
+		// The ClusterClass must exist to proceed with update validation. Return an error if the ClusterClass was
+		// not found.
+		if apierrors.IsNotFound(clusterClassGetErr) {
```

Comment:
```suggestion
		if clusterClassGetErr != nil {
```
WDYT about doing it like this? I know based on the if above it's the same. But this seems a bit easier and more robust